hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/MergeIntoHoodieTableCommand.scala

```diff
@@ -802,27 +813,55 @@ case class MergeIntoHoodieTableCommand(mergeInto: MergeIntoTable) extends Hoodie
       assert(insert.assignments.length <= targetTableSchema.length,
         s"The number of insert assignments[${insert.assignments.length}] must be less than or equal to the " +
           s"targetTable field size[${targetTableSchema.length}]"))
-
+    // Precombine field and primary key field must be present in the assignment clause of all insert actions.
```

Comment:
I assume the insert action should always include all columns, so such check may not be necessary, i.e., missing any column would lead to insert failure already?

Reply:
Let's keep the checks so it's strict.